src/app/shared/services/toastr-facade.service.ts

```diff
@@ -0,0 +1,9 @@
+import { inject, Injectable } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ToastrFacade {
+  public toastr = inject(ToastrService);
```

Comment:
nie do końca teraz działa to jak fasada, bo nadal jesteśmy uzależnieni od zewnętrznej biblioteki, chodzi tutaj o to by dodać konkretne metody takie jak `error` `success`, `warning` itp, które przyjmą taki kształt

```typescript
private toastr = inject(ToastrService);

public error(message, heading) {
   this.toastr.error(message, heading);
}
```

a następnie w aplikacji będziemy odwoływać się już do konkretnych metod, a nie będzie interesowała nas implementacja ich, jedyne czego potrzebujemy to przekazanie odpowienich argumentów, a w przypadku toastów to oczywistymi parametrami jest wiadomosc w toast lub opcjonalnie  naglowek toasta

Reply:
przy czym przy projektowaniu fasady nie musimy implementować od razu wszystkich metod, które chcemy "opakować" w twoim przypadku wystarczy tylko error, jak ktoś będzie chciał użyć np. success, to wtedy to doda